### PR TITLE
[JENKINS-67531] Do not load Pipelines during PlaceholderTask.getCauseOfBlockage in normal scenarios

### PR DESCRIPTION
**Pull request by dwnusbaum**
See [JENKINS-67351](https://issues.jenkins.io/browse/JENKINS-67351) and https://github.com/jenkinsci/workflow-api-plugin/pull/188. Amends https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/185.

One of the stack traces related to JENKINS-67351 (from the jenkinsci-users mailing list) indicated that a build being accessed in `PlaceholderTask.getCauseOfBlockage` was not yet loaded because `FlowExecutionList$ItemListenerImpl.onLoaded` had not executed yet. I did not think this was possible in https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/185#discussion_r762246581, and we do not want to be loading Pipelines while the queue lock is held in normal scenarios.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jglick — APPROVED** [review]

**Comment by dwnusbaum** [comment]
Leaving in draft until https://github.com/jenkinsci/workflow-api-plugin/pull/188 is released.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java [hunk 1]
```diff
@@ -422,6 +423,10 @@ public String getCookie() {
         }
 
         @Override public CauseOfBlockage getCauseOfBlockage() {
+            if (!FlowExecutionList.get().isResumptionComplete()) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:426]: Maybe use this test to gate https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/0a45db4d9db1da6ee5429ed742344f44f99ecd89/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L430-L435 but always do https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/0a45db4d9db1da6ee5429ed742344f44f99ecd89/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L436-L443 though #185 did introduce both blocks.

**Review by jglick — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java (hunk 1 above)
Comment (dwnusbaum) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java]: See ef7ef1d.

**Review by jglick — APPROVED** [review]

**Review by car-roll — APPROVED** [review]

**Comment by KalleOlaviNiemitalo** [comment]
There is a typo in the PR title; it refers to the nonexistent [JENKINS-67531](https://issues.jenkins.io/browse/JENKINS-67531), rather than [JENKINS-67351](https://issues.jenkins.io/browse/JENKINS-67351). This has propagated to the [1107.v5dab75aaccbd](https://github.com/jenkinsci/workflow-durable-task-step-plugin/releases/tag/1107.v5dab75aaccbd) release notes and to the [plugin page](https://plugins.jenkins.io/workflow-durable-task-step/#releases), as well.

**Pull request merged**